### PR TITLE
refactor: initial cleanup of Drive and DPP document types

### PR DESCRIPTION
**Pull request by QuantumExplorer**
There is a lot of code and logic duplicated between Drive and Dpp, this PR tries to reduce it.

## What was done?
Removed a lot of code duplication, and moved things from Drive to DPP, or DPP to drive.

## How Has This Been Tested?
CI + ran locally.

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-drive-abci/src/execution/fee_pools/fee_distribution.rs [hunk 1]
```diff
@@ -874,7 +874,7 @@ mod tests {
             assert!(matches!(
                 proposer_payouts,
                 Some(ProposersPayouts {
-                    proposers_paid_count: proposers_count,
+                    proposers_paid_count: _proposers_count,
```
Comment (shumkov) [inline packages/rs-drive-abci/src/execution/fee_pools/fee_distribution.rs:877]: You need to add a guard to check the value or add `..` instead if you don't care about it

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-dpp/src/data_contract/data_contract.rs [hunk 2]
```diff
@@ -466,6 +473,96 @@ impl TryFrom<Vec<u8>> for DataContract {
     }
 }
 
+pub fn get_mutability(
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-dpp/src/data_contract/data_contract.rs:476]: Okay

File: packages/rs-dpp/src/data_contract/data_contract.rs [hunk 3]
```diff
@@ -371,26 +361,42 @@ impl DataContract {
...
+                    data_contract: self.clone(),
+                },
+            ))
+    }
+
+    /// Returns the binary properties for the given document type
+    /// Comparing to JS version of DPP, the binary_properties are not generated automatically
+    /// if they're not present. It is up to the developer to use proper methods like ['DataContract::set_document_schema'] which
+    /// automatically generates binary properties when setting the Json Schema
+    pub fn get_optional_binary_properties(
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-dpp/src/data_contract/data_contract.rs:377]: I did not write this. I'll add a todo to rename.

File: packages/rs-dpp/src/data_contract/document_type/index.rs [hunk 4]
```diff
@@ -122,50 +161,74 @@ impl Index {
...
+                        let index_property = IndexProperty::from_cbor_value(property_map)?;
+                        index_properties.push(index_property);
+                    }
                 }
+                _ => {}
             }
         }
 
+        // if the index didn't have a name let's make one
+        let name = name.unwrap_or(Alphanumeric.sample_string(&mut rand::thread_rng(), 24));
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-dpp/src/data_contract/document_type/index.rs:228]: We should remove index names in a future PR altogether.

File: packages/rs-dpp/src/data_contract/extra/common.rs [hunk 5]
```diff
@@ -78,18 +124,20 @@ pub fn cbor_inner_array_of_strings<'a>(
...
-) -> Option<&'a Vec<(Value, Value)>> {
-    let key_value = get_key_from_cbor_map(document_type, key)?;
-    if let Value::Map(map_value) = key_value {
-        return Some(map_value);
-    }
-    None
-}
-
+//
+// pub fn cbor_inner_map_value<'a>(
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-dpp/src/data_contract/extra/common.rs:128]: It's part of the phase 2 of the refactor

File: packages/rs-dpp/src/data_contract/extra/errors.rs [hunk 6]
```diff
@@ -1,72 +1,29 @@
...
-    #[error("key wrong type error: {0}")]
-    KeyWrongType(&'static str),
-
-    #[error("key out of bounds error: {0}")]
-    KeyWrongBounds(&'static str),
-
-    #[error("value wrong type error: {0}")]
-    ValueWrongType(&'static str),
-}
+// #[derive(Debug, thiserror::Error)]
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-dpp/src/data_contract/extra/errors.rs:1]: Fixed

File: packages/rs-dpp/src/state_transition/validation/validate_state_transition_key_signature.rs [hunk 7]
```diff
@@ -287,7 +287,7 @@ mod test {
         let TestData {
             mut state_repository,
             asset_lock_public_key_hash_fetcher,
-            bls,
+            bls: _,
```
Comment (shumkov) [inline packages/rs-dpp/src/state_transition/validation/validate_state_transition_key_signature.rs:290]: ..

File: packages/rs-dpp/src/data_contract/extra/common.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-dpp/src/data_contract/extra/common.rs:116]: Yeah, I got it. Usually, we do not allow you to commit commented or dead code but if you really need it you can do it with TODO so you won't forget about it and other people won't be confused seeing it.

**Review by shumkov — DISMISSED** [review]
👍

**Pull request merged**